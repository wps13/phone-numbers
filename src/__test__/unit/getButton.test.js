import React from "react";

import ReactDOM from "react-dom";

import GetButton from "../../components/GetNumbersButton";

import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<GetButton />, div);
});

it("render button correctly", () => {
  const { getByTestId } = render(<GetButton />);
  expect(getByTestId("button")).toHaveTextContent("Get Numbers");
});

it("matches snapshot", () => {
  const tree = renderer.create(<GetButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
