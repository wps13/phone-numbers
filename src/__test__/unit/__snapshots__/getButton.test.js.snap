// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="getNumbers"
  data-testid="button"
  onClick={[Function]}
>
  <p>
    Get Numbers
  </p>
</div>
`;
